### ISSUE TITLE
networking: Failed creating ingress sandbox error on Manager promote and restart

### ISSUE THREAD

**Issue by abronan** [docker/docker#24346]
**Output of `docker version`:**

```
Client:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   ecc3cfe
 Built:        Tue Jul  5 21:26:01 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.25
 Go version:   go1.6.2
 Git commit:   ecc3cfe
 Built:        Tue Jul  5 21:26:01 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.12.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host overlay
Swarm: active
 NodeID: 1qt5hft58p5du9w5cx6zfi52h
 IsManager: Yes
 Managers: 2
 Nodes: 3
 CACertHash: sha256:edf6aa23825bb5a39f9f7bb8708957cff940ccd2aa6bfdf8a9a86e87b8734789
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-28-generic
Operating System: Ubuntu 16.04 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.954 GiB
Name: node01
ID: ZTNM:55LA:E6YV:QC3Y:5QNC:LFMK:L4IE:MQ3Q:S43M:D45D:H4P7:HPLC
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Digital Ocean VMs:

*3 Managers*: 2 GB Memory / 40 GB Disk / Ubuntu 16.04 x64
*3 Agents*: 1 GB Memory / 30 GB Disk / Ubuntu 16.04 x64

**Steps to reproduce the issue:**
1. Start the docker daemon on each machine
2. `swarm init`
3. `swarm node promote <any-node>`

**Describe the results you received:**

The process is successful but I get handful of error logs coming from libnetwork on the promoted node, following is an example:

```
ERRO[0065] Failed creating ingress sandbox: container ingress-sbox is already present: &{e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1 ingress-sbox {{  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/hosts  [] []} {/var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf.hash [] [] []} map[] false false 0 []} [8.8.8.8 8.8.4.4] 0xc821343ad0 0xc8202082d0 0xc820402700 {{0 0} 1} 0 [0xc820337600 0xc8200806e0] map[] map[5fcfab41dd078935e94b86a4f5d11b5f2761a5a7c2da9a2e8cd33a2df92463fa:{} 5cceffade038b78d3f0b438bfb795784071dfb05df41d10953eb3ca76718bad7:{}] <nil> 14 true false false true {0 0}}
```

**Describe the results you expected:**

No error logs.

**Additional information you deem important (e.g. issue happens only occasionally):**

Those logs are from fresh nodes after cleaning `/var/lib/docker`.

Full error log from starting the daemon to the node being promoted:

```
INFO[0000] New containerd process, pid: 4302            
WARN[0000] containerd: low RLIMIT_NOFILE changing to max  current=1024 max=65536
ERRO[0001] devmapper: Udev sync is not supported. This will lead to data loss and unexpected behavior. Install a dynamic binary to use devicemapper or select a different storage driver. For more information, see https://docs.docker.com/engine/reference/commandline/daemon/#daemon-storage-driver-option 
INFO[0001] Graph migration to content-addressability took 0.00 seconds 
WARN[0001] Your kernel does not support swap memory limit. 
INFO[0001] Loading containers: start.                   
INFO[0001] Firewalld running: false                     
INFO[0001] Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address 
INFO[0001] Loading containers: done.                    
INFO[0001] Daemon has completed initialization          
INFO[0001] Docker daemon                                 commit=ecc3cfe graphdriver=overlay version=1.12.0-dev
INFO[0001] API listen on /var/run/docker.sock           
INFO[0036] Waiting for TLS certificate to be issued...  
INFO[0036] Downloaded new TLS credentials with role: swarm-worker. 
INFO[0000] Firewalld running: false                     
INFO[0065] Renewing TLS Certificate.                    
ERRO[0065] Failed creating ingress sandbox: container ingress-sbox is already present: &{e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1 ingress-sbox {{  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/hosts  [] []} {/var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf.hash [] [] []} map[] false false 0 []} [8.8.8.8 8.8.4.4] 0xc821343ad0 0xc8202082d0 0xc820402700 {{0 0} 1} 0 [0xc820337600 0xc8200806e0] map[] map[5fcfab41dd078935e94b86a4f5d11b5f2761a5a7c2da9a2e8cd33a2df92463fa:{} 5cceffade038b78d3f0b438bfb795784071dfb05df41d10953eb3ca76718bad7:{}] <nil> 14 true false false true {0 0}} 
INFO[0065] Waiting for TLS certificate to be issued...  
ERRO[0065] Failed creating ingress sandbox: container ingress-sbox is already present: &{e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1 ingress-sbox {{  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/hosts  [] []} {/var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf.hash [] [] []} map[] false false 0 []} [8.8.8.8 8.8.4.4] 0xc821343ad0 0xc8202082d0 0xc820402700 {{0 0} 1} 0 [0xc820337600 0xc8200806e0] map[] map[5fcfab41dd078935e94b86a4f5d11b5f2761a5a7c2da9a2e8cd33a2df92463fa:{} 5cceffade038b78d3f0b438bfb795784071dfb05df41d10953eb3ca76718bad7:{}] <nil> 14 true false false true {0 0}} 
ERRO[0065] Failed creating ingress sandbox: container ingress-sbox is already present: &{e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1 ingress-sbox {{  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/hosts  [] []} {/var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf  /var/lib/docker/network/files/e9fd64e63d572cdc5726177131bc9897d3b838c34f394fa32c7e7eb2f2d6c5f1/resolv.conf.hash [] [] []} map[] false false 0 []} [8.8.8.8 8.8.4.4] 0xc821343ad0 0xc8202082d0 0xc820402700 {{0 0} 1} 0 [0xc820337600 0xc8200806e0] map[] map[5fcfab41dd078935e94b86a4f5d11b5f2761a5a7c2da9a2e8cd33a2df92463fa:{} 5cceffade038b78d3f0b438bfb795784071dfb05df41d10953eb3ca76718bad7:{}] <nil> 14 true false false true {0 0}} 
INFO[0065] Downloaded new TLS credentials with role: swarm-worker. 
INFO[0065] Listening for local connections               addr=/var/lib/docker/swarm/control.sock proto=unix
INFO[0065] Listening for connections                     addr=X.X.X.X:2377 proto=tcp
INFO[0066] 457ac3c847bed54a became follower at term 0   
INFO[0066] newRaft 457ac3c847bed54a [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0] 
INFO[0066] 457ac3c847bed54a became follower at term 1   
INFO[0066] 457ac3c847bed54a [term: 1] received a MsgHeartbeat message with higher term from 687c9a24ab61cccd [term: 2] 
INFO[0066] 457ac3c847bed54a became follower at term 2   
INFO[0066] raft.node: 457ac3c847bed54a elected leader 687c9a24ab61cccd at term 2
```

/cc @mrjana @sanimej

**Comment by abronan** [comment]
Update: The failure only happens when I first join a node as an Agent and later promote it as a Manager. This does not happen when I join a node directly as a Manager using `join --manager`.

**Comment by thaJeztah** [comment]
ping @mrjana @sanimej PTAL

**Comment by cpuguy83** [comment]
Closed by #24573
